st2api/st2api/controllers/v1/actionviews.py

```diff
@@ -98,32 +95,36 @@ class OverviewController(resource.ContentPackResourceController):
 
     include_reference = True
 
-    @jsexpose(arg_types=[str])
     def get_one(self, ref_or_id):
         """
             List action by id.
 
             Handle:
                 GET /actions/views/overview/1
         """
-        action_api = super(OverviewController, self)._get_one(ref_or_id)
-        return self._transform_action_api(action_api)
+        resp = super(OverviewController, self)._get_one(ref_or_id)
+        action_api = resp.json
+        result = self._transform_action_api(action_api)
+        resp.json = result
+        return resp
 
-    @jsexpose(arg_types=[str])
     def get_all(self, **kwargs):
         """
             List all actions.
 
             Handles requests:
                 GET /actions/views/overview
         """
-        action_apis = super(OverviewController, self)._get_all(**kwargs)
-        return map(self._transform_action_api, action_apis)
+        resp = super(OverviewController, self)._get_all(**kwargs)
+        action_apis = resp.json
+        result = map(self._transform_action_api, action_apis)
+        resp.json = result
+        return resp
 
     @staticmethod
     def _transform_action_api(action_api):
-        action_id = action_api.id
-        action_api.parameters = ParametersViewController._get_one(action_id).get('parameters')
+        action_id = action_api['id']
```

Comment:
kami After reading enykeev's explanation and looking at the code, I think that _get_all() should really return classes and not JSON. I think I prefer to work with classes than dicts as kami outlined and I am also of the opinion that class.to_json() should be implemented. The serialization and deserialization from/to JSON should happen only at the HTTP interface/handler.

>Although we still support controllers returning models, I find this practice highly detrimental to us better understanding how http stack works. Which was part of the reason I'm doing it to begin with. 

Is the problem because of our convoluted code or a design issue? In previous jobs etc, what typically happened was the API payload was validated against and a schema. You would have another library that converted these objects to classes and you'd have a from_json and to_json method defined on the classes. Any code that worked with these objects dealt only with the models. The JSON serialization/deserialization is restricted to HTTP handler. We never had any confusions when we did that. I've seen this pattern at multiple places in different languages.